crates/eips/src/eip7702/auth_list.rs

```diff
@@ -225,9 +238,12 @@ impl RecoveredAuthorization {
         Self { inner, authority }
     }
 
-    /// Get the `authority` for the authorization.
-    pub fn authority(&self) -> RecoveredAuthority {
-        self.authority.clone()
+    /// Returns an optional address based on the current state of the authority.
+    pub fn authority(&self) -> Option<Address> {
+        match &self.authority {
+            RecoveredAuthority::Valid(address) => Some(address.clone()),
+            RecoveredAuthority::Invalid => None,
+        }
```

Comment:
this should just call self.authority.address

Reply:
> this should just call self.authority.address

Done